Firestore/core/src/firebase/firestore/remote/grpc_stream.cc

```diff
@@ -98,6 +98,7 @@ GrpcStream::GrpcStream(
 GrpcStream::~GrpcStream() {
   HARD_ASSERT(completions_.empty(),
               "GrpcStream is being destroyed without proper shutdown");
+  MaybeUnregister();
```

Comment:
The new tests exposed a bug when under certain code paths, `GrpcConnection::Unregister` will not be called. To make it more robust, moved unregistering into the destructor (here and other `GrpcCall`-derived classes).